src/test/java/com/karankumar/bookproject/backend/entity/BookTest.java

```diff
@@ -0,0 +1,84 @@
+package com.karankumar.bookproject.backend.entity;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.service.BookService;
+import com.karankumar.bookproject.backend.service.PredefinedShelfService;
+import com.karankumar.bookproject.backend.service.TagService;
+import com.karankumar.bookproject.backend.utils.PredefinedShelfUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Bo Jacobs
+ */
+
+@IntegrationTest
+public class BookTest {
+
+    private static BookService bookService;
+    private static TagService tagService;
+
+    private static Book testBook;
+    private static Tag testTag;
+
+    @BeforeAll
```

Comment:
Could you please change this to a `@BeforeEach`? This can then be limited to resetting the book and tag service. 

This will ensure that, if new tests are added, they do not interfere with one another.

Reply:
The autowiring and initialisation of fields can be moved to a constructor so that it still only happens once